src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -725,15 +738,13 @@ void SHAMap::getFetchPack (SHAMap* have, bool includeLeaves, int max,
 
                 if (next->isInner ())
                 {
-                    if (!have || !have->hasInnerNode (childID, childHash))
+                    if (! have || ! have->hasInnerNode (childID, childHash))
                         stack.push ({next, childID});
                 }
-                else if (includeLeaves && (!have || !have->hasLeafNode (next->peekItem()->getTag(), childHash)))
+                else if (! have || ! have->hasLeafNode (next->peekItem()->getTag(), childHash))
```

Comment:
Very nice refactor, reusing `func` three times and passing it in externally.  As far as I can see all of the pre-existing behavior is preserved with one exception.  We're no longer checking `includeLeaves` in this `else if`.  I suspect that's intentional, but I thought I'd ask to make sure.